### PR TITLE
Issue-1-create-an-entry - Jimmy Vu

### PR DESCRIPTION
**Pull request by Jimmy-Vu**
https://user-images.githubusercontent.com/88172055/132565686-749fc65a-5cba-4f98-a64e-08f0d8f66097.mov

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+h3 {
+  margin-bottom: 10px;
+  font-weight: 400;
+}
+
+.header-banner {
+  background-color: #572b82;
+  width: 100%;
```
Comment (codingcodymiller) [inline css/styles.css]: Why not just use a `column-full` class rather than manually setting `width`?

File: css/styles.css [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+.header-title {
+  color: white;
+  max-width: 900px;
```
Comment (codingcodymiller) [inline css/styles.css]: You should just use the `container` class

File: index.html [hunk 3]
```diff
@@ -13,11 +13,38 @@
...
+    <header class="header-title">
+      <h2>Code Journal</h2>
+    </header>
+  </div>
   <main>
+    <div data-view="entry-form" class="container">
+      <div class="row">
+          <form id="form" class="column-full">
+            <h1>New Entry</h1>
+            <div class="display-flex">
```
Comment (codingcodymiller) [inline index.html]: Why not use `row`?

File: index.html [hunk 4]
```diff
@@ -13,11 +13,38 @@
...
+              <img id="main-image" src="images/placeholder-image-square.jpg">
```
Comment (codingcodymiller) [inline index.html]: This should be half the available width in the row, you should probably use the `column-half` class

File: index.html [hunk 5]
```diff
@@ -13,11 +13,38 @@
...
+              <div class="column-full display-inline">
```
Comment (codingcodymiller) [inline index.html]: Same here, this should probably be `column-half`, not `column-full`

File: index.html [hunk 6]
```diff
@@ -13,11 +13,38 @@
...
+                <h3 class="no-margin-top">Title</h3>
+                <input id="title-text" type="text" required>
+                <h3>Photo URL</h3>
+                <input id="input-url" type="text" required>
+              </div>
+            </div>
+            <div class="display-flex-column">
```
Comment (codingcodymiller) [inline index.html]: How about just `column-full`? Why does this need to be a column flexbox?

File: js/main.js [hunk 7]
```diff
@@ -1,2 +1,41 @@
...
+  }
+}
+
+urlInput.addEventListener('input', updatePhoto);
+
+var form = document.querySelector('#form');
+var title = document.querySelector('#title-text');
+var notes = document.querySelector('#notes-text');
+var savedJournal = [];
+var entryID = 0;
```
Comment (codingcodymiller) [inline js/main.js]: There should be no need for these variables, you should just be using your data model stored in the `data` variable in `data.js`

File: js/main.js [hunk 8]
```diff
@@ -1,2 +1,41 @@
...
+
+  savedJournal.unshift({
+    title: title.value,
+    photoURL: urlInput.value,
+    notes: notes.value,
+    entryID: entryID
+  });
+  entryID++;
+  var jsonJournal = JSON.stringify(savedJournal);
+  localStorage.setItem('journalEntry', jsonJournal);
```
Comment (codingcodymiller) [inline js/main.js]: All `localStorage` related code should be in `data.js` so that it is in the same file as the data it is preserving.

Comment (codingcodymiller) [reply js/main.js]: Also, there is no need to save data in localStorage after every submitted entry, you just need to do it before the page unloads. I encourage you to look back at `javascript-local-storage`, the code you'll need for this project will be very similar to the code you wrote in that exercise.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
In addition to my other comments, I don't see you checking if there is a value in `localStorage` when the page loads. You should be checking if there is a value there, and if there is, replacing the current value of `data` with that object.

File: index.html [hunk 9]
```diff
@@ -28,19 +28,21 @@ <h2>Code Journal</h2>
       <div class="row">
           <form id="form" class="column-full">
             <h1>New Entry</h1>
-            <div class="display-flex">
-              <img id="main-image" src="images/placeholder-image-square.jpg">
-              <div class="column-full display-inline">
+            <div class="row">
+              <img id="main-image" class="column-full" src="images/placeholder-image-square.jpg">
```
Comment (codingcodymiller) [inline index.html]: Shouldn't this be `column-half`? The image only takes up half of the row on desktop.

File: js/data.js [hunk 10]
```diff
@@ -6,3 +6,39 @@ var data = {
...
+  });
+  data.nextEntryId++;
+
+  photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  title.value = '';
+  urlInput.value = '';
+  notes.value = '';
+}
+
+form.addEventListener('submit', submitListener);
```
Comment (codingcodymiller) [inline js/data.js]: This should be in the `main.js` file. `data.js` is only for the data model and localStorage code.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/data.js (hunk 10 above)
Comment (Jimmy-Vu) [reply js/data.js]: This was my mistake for misunderstanding your earlier feedback.

File: js/main.js [hunk 11]
```diff
@@ -12,3 +12,25 @@ function updatePhoto(event) {
...
+    photoURL: urlInput.value,
+    notes: notes.value,
+    entryID: data.nextEntryId
+  });
+  data.nextEntryId++;
+
+  photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  title.value = '';
+  urlInput.value = '';
+  notes.value = '';
```
Comment (codingcodymiller) [inline js/main.js]: I would just use the `reset` method

https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reset

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**